### PR TITLE
(docs) - Debugging Documentation

### PR DESCRIPTION
**Pull request by andyrichardson**
## About

Add new debugging docs!

[Markdown](https://github.com/FormidableLabs/urql/blob/add-debugging-docs/docs/advanced/debugging.md)

### CODE REVIEW

File: docs/advanced/debugging.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
+---
+title: Debugging
+order: 6
+---
+
+# Debugging
+
+We've tried to make debugging in `urql` as seamless as possible by creating tools for both basic users and those trying to create their own exchanges.
```
Comment (JoviDeCroock) [inline docs/advanced/debugging.md:8]:
```suggestion
We've tried to make debugging in `urql` as seamless as possible by creating tools for our users using urql and our users creating their own exchanges.
```

I'd avoid "basic" has a negative co-notation imo

File: docs/advanced/debugging.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+## Debug events
+
+Debug events are a means for seeing what's going on inside of Urql exchanges. Before getting started, here are a few things to be aware of.
```
Comment (JoviDeCroock) [inline docs/advanced/debugging.md:12]:
```suggestion
Debug events are a means for seeing what's going on inside of urql exchanges. Before getting started, here are a few things to be aware of.
```

File: docs/advanced/debugging.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+This includes the creator of an `urql` client and any of it's exchanges (although the latter is not advised).
+
+### Debug events are disabled in production
+
+Debug events are a fire-and-forget mechanism and should not be used as a means for messaging in your app.
+
+## Consuming debug events
+
+Debugging events can be inspected either graphically using our devtools or manually by subscribing to events on the client.
```
Comment (JoviDeCroock) [inline docs/advanced/debugging.md:24]:
```suggestion
Debugging events can be inspected either graphically using our [devtools](https://github.com/FormidableLabs/urql-devtools) or manually by subscribing to events on the client.
```

File: docs/advanced/debugging.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+### Devtools
+
+The quickest way is going to be using our [devtools extension](https://github.com/FormidableLabs/urql-devtools/) which visualizes events on a timeline and provides tools to filter events, inspect the cache, and trigger custom querys via the running client.
```
Comment (JoviDeCroock) [inline docs/advanced/debugging.md:28]: Not sure but is the plural of query, queries?

File: docs/advanced/debugging.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+// Unsubscribe from events
+unsubscribe();
+```
+
+## Dispatching debug events
+
+Debug events are a means of sharing implementation details to consumers of an exchange.
+
+### Dispatching a debug event
+
```
Comment (JoviDeCroock) [inline docs/advanced/debugging.md:61]: We should give this title some content else we should delete it, no?

Comment (andyrichardson) [reply docs/advanced/debugging.md:61]: Good spot!

File: docs/advanced/debugging.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+// ...
+// Unsubscribe from events
+unsubscribe();
+```
+
+## Dispatching debug events
+
+Debug events are a means of sharing implementation details to consumers of an exchange.
+
+### Dispatching a debug event
```
Comment (andyrichardson) [inline docs/advanced/debugging.md:60]:
```suggestion
```

File: docs/advanced/debugging.md (hunk 4 above)
Comment (andyrichardson) [reply docs/advanced/debugging.md:28]: Yeah my bad!
```suggestion
The quickest way is going to be using our [devtools extension](https://github.com/FormidableLabs/urql-devtools/) which visualizes events on a timeline and provides tools to filter events, inspect the cache, and trigger custom queries via the running client.
```

**Pull request merged**